src/leveled_sst.erl

```diff
@@ -1946,45 +1948,51 @@ lookup_slots(StartKey, EndKey, Tree, FilterFun) ->
 %% based on a 17-bit hash (so 0.0039 fpr).
 
 
--spec accumulate_positions(leveled_codec:ledger_kv(),
-                            {binary(),
-                                non_neg_integer(),
-                                list(non_neg_integer()),
-                                leveled_codec:last_moddate()}) ->
-                                    {binary(),
-                                        non_neg_integer(),
-                                        list(non_neg_integer()),
-                                        leveled_codec:last_moddate()}.
+-spec accumulate_positions(
+        list(leveled_codec:ledger_kv()),
+        {binary(),
+            non_neg_integer(),
+            list(leveled_codec:segment_hash()),
+            leveled_codec:last_moddate()}) ->
+                {binary(),
+                    non_neg_integer(),
+                    list(leveled_codec:segment_hash()),
+                    leveled_codec:last_moddate()}.
 %% @doc
 %% Fold function use to accumulate the position information needed to
 %% populate the summary of the slot
-accumulate_positions({K, V}, {PosBinAcc, NoHashCount, HashAcc, LMDAcc}) ->
+accumulate_positions([], Acc) ->
+    Acc;
+accumulate_positions([{K, V}|T], {PosBin, NoHashCount, HashAcc, LMDAcc}) ->
     {_SQN, H1, LMD} = leveled_codec:strip_to_indexdetails({K, V}),
     LMDAcc0 = take_max_lastmoddate(LMD, LMDAcc),
-    PosH1 = extract_hash(H1),
-    case is_integer(PosH1) of
-        true ->
+    case extract_hash(H1) of
+        PosH1 when is_integer(PosH1) ->
             case NoHashCount of
                 0 ->
-                    {<<1:1/integer, PosH1:15/integer,PosBinAcc/binary>>,
-                        0,
-                        [H1|HashAcc],
-                        LMDAcc0};
+                    accumulate_positions(
+                        T,
+                        {<<(32768 + PosH1):16/integer, PosBin/binary>>,
```

Comment:
This is correct, but I found the previous code more readable. In order to validate correctness, one really needs to know what extract_hash is doing, whereas in the earlier code it was clear that a new integer was created, no matter what 16 bit PosH1 was provided.

Reply:
I've changed the tune_hash/1 function to return a 2-byte hash between 2 ^ 15 and 2 ^16 - 1, to make this a bit more explicit.

I'd seen some (inconclusive) evidence from profiling that there was an improvement when working in full bytes.  As part of the change to using a 2-byte hash, the find_pos function has been refactored to match instead on whole bytes.

The combination of this and splitting the find_pos function to be different when matching on multiple or single hashes - improved the time taken by find_pos in both Head load and AAE fold load (noticeably in the latter by > 10%).